[WIP] Add 3d import and export

cymed opened the pull request:
This PR aims at adding import and export of 3D INTERLIS models

cymed committed:
add export

cymed committed:
add import

cymed committed:
add cover _depth

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

cymed committed:
fix

cymed committed:
gui & cmd

cymed committed:
cover._depth trigger funcs

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

cymed committed:
fix

sjib commented:
Do we have a clear understanding which is the master level? 

- reach_point.level 
- or the start / end of reach_progression3d

cymed committed:
fix

cymed commented:
> Maechtigkeit should always be a function of the levels on export and not be considered on import to overwrite the levels.

Maechtigkeit is not imported. On export, I opted for storing the value on the db and updating through triggers instead of on-the-fly calculations on export

cymed committed:
fix

cymed committed:
Update geometry_functions.sql

cymed committed:
semicolon

cymed committed:
move loop

cymed committed:
test

cymed committed:
separate functions

cymed committed:
fix

cymed committed:
semicolons

cymed committed:
Update geometry_functions.sql

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

cymed committed:
Update geometry_functions.sql

cymed committed:
Update geometry_functions.sql

cymed committed:
Update geometry_functions.sql

cymed committed:
Update geometry_functions.sql

cymed committed:
Update geometry_functions.sql

cymed committed:
Update geometry_functions.sql

cymed committed:
typo

cymed committed:
fix detail geometry and prepare test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

sjib commented on plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py:
```diff
@@ -32,6 +32,12 @@ def __init__(self, parent):
             config.MODEL_NAME_SIA405_BASE_ABWASSER, [config.MODEL_NAME_SIA405_BASE_ABWASSER]
         )
+        self.export_model_selection_comboBox.addItem(
+            config.MODEL_NAME_DSS_3D, [config.MODMODEL_NAME_DSS_3DEL_NAME_DSS]
```
Should be:
```
        self.export_model_selection_comboBox.addItem(
            config.MODEL_NAME_DSS_3D, [config.MODEL_NAME_DSS_3D]
        )
```

sjib replied:
Then dialog shows correctly:
<img width="485" height="548" alt="grafik" src="https://github.com/user-attachments/assets/2ed7d6e0-d1d5-4b14-a86d-d59e140fcf6c" />

cymed committed:
fix

sjib commented:
```
Ein Fehler trat bei der Ausführung von Python-Code auf: 

TypeError: 'NoneType' object is not callable 
Traceback (most recent call last):
  File "C:\Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python/plugins\teksi_wastewater\teksi_wastewater_plugin.py", line 637, in actionExportClicked
    self.interlisImporterExporter.action_export()
  File "C:\Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python/plugins\teksi_wastewater\interlis\gui\interlis_importer_exporter_gui.py", line 163, in action_export
    raise exception
  File "C:\Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python/plugins\teksi_wastewater\interlis\gui\interlis_importer_exporter_gui.py", line 131, in action_export
    self.interlis_importer_exporter.interlis_export(
  File "C:\Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python/plugins\teksi_wastewater\interlis\interlis_importer_exporter.py", line 637, in interlis_export
    self._export_to_intermediate_schema(
  File "C:\Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python/plugins\teksi_wastewater\interlis\interlis_importer_exporter.py", line 848, in _export_to_intermediate_schema
    self._init_model_classes(export_model)
  File "C:\Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python/plugins\teksi_wastewater\interlis\interlis_importer_exporter.py", line 2299, in _init_model_classes
    self.model_classes_interlis = ModelInterlis().classes()
                                  ^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not callable


Python-Version: 3.12.10 (main, Apr  9 2025, 08:14:47) [MSC v.1943 64 bit (AMD64)] 
QGIS-Version: 3.40.6-Bratislava Bratislava, 5d7f6a3e26 

Python-Pfad:
C:/PROGRA~1/QGIS34~1.6/apps/qgis-ltr/./python
C:/Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python
C:/Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python/plugins
C:/PROGRA~1/QGIS34~1.6/apps/qgis-ltr/./python/plugins
C:\PROGRA~1\QGIS34~1.6\apps\grass\grass84\etc\python
C:\Users\stefa\Documents
C:\Program Files\QGIS 3.40.6\bin\python312.zip
C:\PROGRA~1\QGIS34~1.6\apps\Python312\DLLs
C:\PROGRA~1\QGIS34~1.6\apps\Python312\Lib
C:\Program Files\QGIS 3.40.6\bin
C:\Users\stefa\AppData\Roaming\Python\Python312\site-packages
C:\PROGRA~1\QGIS34~1.6\apps\Python312
C:\PROGRA~1\QGIS34~1.6\apps\Python312\Lib\site-packages
C:\PROGRA~1\QGIS34~1.6\apps\Python312\Lib\site-packages\win32
C:\PROGRA~1\QGIS34~1.6\apps\Python312\Lib\site-packages\win32\lib
C:\PROGRA~1\QGIS34~1.6\apps\Python312\Lib\site-packages\Pythonwin
C:/Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python
C:/Daten/Testing/TEKSI/2025_0_2rc7_interils3d
C:\Users/stefa/AppData/Roaming/QGIS/QGIS3\profiles\2025_0_2rc7_interils3d/python/plugins\teksi_wastewater\interlis\gui\..\..
.
```

Here the 3D models are missing:

```
    def _init_model_classes(self, model):
        ModelInterlis = None
        if model == config.MODEL_NAME_AG96:
            ModelInterlis = ModelInterlisAG96
        elif model == config.MODEL_NAME_AG64:
            ModelInterlis = ModelInterlisAG64
        elif model == config.MODEL_NAME_SIA405_BASE_ABWASSER:
            ModelInterlis = ModelInterlisSia405BaseAbwasser
        elif model == config.MODEL_NAME_SIA405_ABWASSER:
            ModelInterlis = ModelInterlisSia405Abwasser
        elif model == config.MODEL_NAME_DSS:
            ModelInterlis = ModelInterlisDss
        elif model == config.MODEL_NAME_VSA_KEK:
            ModelInterlis = ModelInterlisVsaKek
        self.model_classes_interlis = ModelInterlis().classes()
        self._progress_done(self.current_progress + 1)
```

cymed committed:
add to model selection dialog

We need no new model definition as they only extend existing models, not creating new classes

sjib commented:
plugins\teksi_wastewater\interlis\interlis_model_mapping\interlis_exporter_to_intermediate_schema.py", line **100**, in tww_export

sjib commented:
TypeError: 'detailgeometrie3d' is an invalid keyword argument for kanal

sjib committed:
typo detaildetailgeometrie3d

sjib commented:
I am trying an export: SIA405_ABWASSER_3D

```
2025-10-24T13:13:06        teksi_wastewater:TEKSI Wastewater plugin version 0.0.0 started
2025-10-24T13:13:06        teksi_wastewater.utils.translation:Using system default locale: de_CH
2025-10-24T13:13:19        teksi_wastewater:dataprovider of vw_tww_wastewater_structure: 
2025-10-24T13:13:19        teksi_wastewater.utils:Checking setup of oid prefix
2025-10-24T13:13:19        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:19        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:19        teksi_wastewater.utils:Checking setup of default_values
2025-10-24T13:13:19        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:19        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:19        teksi_wastewater.utils:Check symbology triggers enabled
2025-10-24T13:13:19        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:19        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:46        teksi_wastewater.utils:Debug.print export_model 'SIA405_ABWASSER_3D_2020_1_LV95'
2025-10-24T13:13:46        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:46        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:46        teksi_wastewater.utils:-----
2025-10-24T13:13:46        teksi_wastewater.utils:INTEGRITY CHECK missing identifiers...
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'data_media' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'file' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'maintenance_event' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'organisation' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'wastewater_structure' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'wastewater_networkelement' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'structure_part' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'reach_point' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'pipe_profile' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'building_group' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'catchment_area' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'connection_object' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'control_center' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'hydr_geometry' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'hydraulic_char_data' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'measure' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'measurement_result' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'measurement_series' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'measuring_device' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'measuring_point' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'mechanical_pretreatment' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'overflow' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'overflow_char' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'retention_body' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'sludge_treatment' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'surface_runoff_parameters' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'throttle_shut_off_unit' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'waste_water_treatment' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'waste_water_treatment_plant' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'wwtp_energy_use' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:No rows in class 'zone' without identifier
2025-10-24T13:13:46        teksi_wastewater.utils:missing_identifier_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:OK: all identifiers set in tww_od.zone!
2025-10-24T13:13:46        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:46        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:46        teksi_wastewater.utils:-----
2025-10-24T13:13:46        teksi_wastewater.utils:INTEGRITY CHECK missing MANDATORY owner references fk_owner...
2025-10-24T13:13:46        teksi_wastewater.utils:Number of datasets in class 'wastewater_structure' without fk_owner : 0
2025-10-24T13:13:46        teksi_wastewater.utils:missing_fk_owner_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:OK: all mandatory fk_owner set in tww_od.wastewater_structure!
2025-10-24T13:13:46        teksi_wastewater.utils:missing_fk_owner_count : 0
2025-10-24T13:13:46        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:46        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing MANDATORY operator references fk_operator...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'wastewater_structure' without fk_operator : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_operator_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_operator set in tww_od.wastewater_structure!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing dataowner references fk_dataowner...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'damage' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'data_media' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'file' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'maintenance_event' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'wastewater_structure' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'wastewater_networkelement' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'structure_part' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'reach_point' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'pipe_profile' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'building_group' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'building_group_baugwr' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'catchment_area' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'connection_object' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'control_center' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'disposal' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'farm' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hq_relation' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hydraulic_char_data' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hydr_geometry' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hydr_geom_relation' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'log_card' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measure' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measurement_result' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measurement_series' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measuring_device' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measuring_point' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'mechanical_pretreatment' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'mutation' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'overflow' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'overflow_char' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'profile_geometry' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'retention_body' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'sludge_treatment' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'surface_runoff_parameters' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'throttle_shut_off_unit' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'waste_water_treatment' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'waste_water_treatment_plant' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'wwtp_energy_use' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'zone' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'sia405cc_cable' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'sia405cc_cable_point' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'sia405pt_protection_tube' without fk_dataowner : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_dataowner_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_dataowner set in tww_od.sia405pt_protection_tube!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing provider references fk_provider...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'damage' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'data_media' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'file' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'maintenance_event' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'wastewater_structure' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'wastewater_networkelement' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'structure_part' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'reach_point' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'pipe_profile' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'building_group' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'building_group_baugwr' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'catchment_area' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'connection_object' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'control_center' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'disposal' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'farm' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hq_relation' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hydraulic_char_data' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hydr_geometry' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hydr_geom_relation' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'log_card' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measure' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measurement_result' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measurement_series' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measuring_device' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measuring_point' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'mechanical_pretreatment' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'mutation' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'overflow' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'overflow_char' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'profile_geometry' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'retention_body' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'sludge_treatment' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'surface_runoff_parameters' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'throttle_shut_off_unit' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'waste_water_treatment' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'waste_water_treatment_plant' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'wwtp_energy_use' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'zone' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'sia405cc_cable' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'sia405cc_cable_point' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'sia405pt_protection_tube' without fk_provider : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_provider_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_provider set in tww_od.sia405pt_protection_tube!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing wastewater_structure references fk_wastewater_structure...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'structure_part' without fk_wastewater_structure : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_wastewater_structure_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_wastewater_structure set in tww_od.structure_part!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing wastewater_node references fk_wastewater_node...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'hydraulic_char_data' without fk_wastewater_node : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_wastewater_node_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'overflow' without fk_wastewater_node : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_wastewater_node_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'throttle_shut_off_unit' without fk_wastewater_node : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_wastewater_node_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_wastewater_node set in tww_od.throttle_shut_off_unit!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing responsible_entity references fk_responsible_entity...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measure' without fk_responsible_entity : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_responsible_entity_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_responsible_entity set in tww_od.measure!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing responsible_start references fk_responsible_start...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'measure' without fk_responsible_start : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_responsible_start_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_responsible_start set in tww_od.measure!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing discharge_point references fk_discharge_point...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'catchment_area_totals' without fk_discharge_point : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_discharge_point_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_discharge_point set in tww_od.catchment_area_totals!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing hydraulic_char_data references fk_hydraulic_char_data...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'catchment_area_totals' without fk_hydraulic_char_data : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_hydraulic_char_data_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_hydraulic_char_data set in tww_od.catchment_area_totals!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing building_group references fk_building_group...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'building_group_baugwr' without fk_building_group : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_building_group_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_building_group set in tww_od.building_group_baugwr!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing reach references fk_reach...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'reach_progression_alternative' without fk_reach : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_reach_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_reach set in tww_od.reach_progression_alternative!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing reach_point_from references fk_reach_point_from...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'reach' without fk_reach_point_from : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_reach_point_from_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_reach_point_from set in tww_od.reach!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing reach_point_to references fk_reach_point_to...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'reach' without fk_reach_point_to : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_reach_point_to_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_reach_point_to set in tww_od.reach!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing pwwf_wastewater_node references fk_pwwf_wastewater_node...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'log_card' without fk_pwwf_wastewater_node : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_pwwf_wastewater_node_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_pwwf_wastewater_node set in tww_od.log_card!
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:-----
2025-10-24T13:13:47        teksi_wastewater.utils:INTEGRITY CHECK missing catchment_area references fk_catchment_area...
2025-10-24T13:13:47        teksi_wastewater.utils:Number of datasets in class 'surface_runoff_parameters' without fk_catchment_area : 0
2025-10-24T13:13:47        teksi_wastewater.utils:missing_fk_catchment_area_count : 0
2025-10-24T13:13:47        teksi_wastewater.utils:OK: all mandatory fk_catchment_area set in tww_od.surface_runoff_parameters!
2025-10-24T13:13:47        teksi_wastewater.utils:CONNECTING TO DATABASE...
2025-10-24T13:13:47        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:47        teksi_wastewater.utils:psycopg dsn: host=localhost port=5434 user=postgres password=[PASSWORD]
2025-10-24T13:13:47        teksi_wastewater.utils:Truncating all tables in schema pg2ili_abwasser
2025-10-24T13:13:51        teksi_wastewater.utils:Deleting all tables in schema pg2ili_abwasser 
2025-10-24T13:13:51        teksi_wastewater.utils:ILIDB SCHEMAIMPORT INTO pg2ili_abwasser...
2025-10-24T13:13:51        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:51        teksi_wastewater.utils:EXECUTING: "C:\Program Files\OpenLogic\jre-22.0.2.9-hotspot\bin\java.exe" -jar "C:\Users\stefa\AppData\Roaming\QGIS\QGIS3\profiles\2025_0_2rc7_interils3d\python\plugins\teksi_wastewater\libs\modelbaker\iliwrapper\bin\ili2pg-5.3.1\ili2pg-5.3.1.jar" --schemaimport --dbhost "localhost" --dbport "5434" --dbusr "postgres" --dbpwd "[PASSWORD]" --dbdatabase "2025_0_2rc7_interils3d" --dbschema pg2ili_abwasser --createGeomIdx --createFk --createFkIdx --createTidCol --importTid --noSmartMapping --defaultSrsCode 2056 --log "C:\Users\stefa\AppData\Local\Temp\tww2ili\251024131351.ili2pg-schemaimport.log" --nameLang de --models "SIA405_ABWASSER_3D_2020_1_LV95"
2025-10-24T13:13:59        teksi_wastewater.utils:ILIDB SCHEMAIMPORT INTO pg2ili_abwasser...
2025-10-24T13:13:59        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:13:59        teksi_wastewater.utils:EXECUTING: "C:\Program Files\OpenLogic\jre-22.0.2.9-hotspot\bin\java.exe" -jar "C:\Users\stefa\AppData\Roaming\QGIS\QGIS3\profiles\2025_0_2rc7_interils3d\python\plugins\teksi_wastewater\libs\modelbaker\iliwrapper\bin\ili2pg-5.3.1\ili2pg-5.3.1.jar" --schemaimport --dbhost "localhost" --dbport "5434" --dbusr "postgres" --dbpwd "[PASSWORD]" --dbdatabase "2025_0_2rc7_interils3d" --dbschema pg2ili_abwasser --createGeomIdx --createFk --createFkIdx --createTidCol --importTid --noSmartMapping --defaultSrsCode 2056 --log "C:\Users\stefa\AppData\Local\Temp\tww2ili\251024131359.ili2pg-schemaimport.log" --nameLang de --models "SIA405_ABWASSER_3D_2020_1_LV95"
2025-10-24T13:14:06        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:14:07        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:14:09        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:14:10        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:14:10        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:14:10        teksi_wastewater.utils:PGSYSCONFDIR: C:\Daten\Testing\pgconf2025\pg_service.conf
2025-10-24T13:14:10        teksi_wastewater.utils:Exporting TWW.organisation -> ABWASSER.organisation
2025-10-24T13:14:11        teksi_wastewater.utils:done
2025-10-24T13:14:11        teksi_wastewater.utils:Exporting TWW.channel -> ABWASSER.kanal
```

cymed committed:
Merge remote-tracking branch 'upstream/main' into add_3d_export

cymed committed:
Merge branch 'improve-check-verbosity' into add_3d_export

cymed committed:
add separate model definitions

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

sjib commented:
`plugin/teksi_wastewater/interlis/interlis_importer_exporter.py:63:9: F841 local variable 'model_dict' is assigned to but never used`

cymed committed:
fix ci and export mistakes

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

sjib commented:
```
source = '/usr/src/plugin/teksi_wastewater/tests/data/test-dataset-DSS_3D.xtf'
parser = None

    def parse(self, source, parser=None):
        """Load external XML document into element tree.
    
        *source* is a file name or file object, *parser* is an optional parser
        instance that defaults to XMLParser.
    
        ParseError is raised if the parser fails to parse the document.
    
        Returns the root element of the given source document.
    
        """
        close_source = False
        if not hasattr(source, "read"):
>           source = open(source, "rb")
E           FileNotFoundError: [Errno 2] No such file or directory: '/usr/src/plugin/teksi_wastewater/tests/data/test-dataset-DSS_3D.xtf'
```

sjib commented:
```
       if not import_model:
            error_text = f"No supported model was found among '{import_models}'."
            if len(import_models) == 1:
                error_text = f"The model '{import_models[0]}' is not supported."
>           raise InterlisImporterExporterError("Import error", error_text, None)
E           teksi_wastewater.interlis.utils.various.InterlisImporterExporterError: ('Import error', "The model 'VSA_KEK_2020_1_LV95' is not supported.", None)

teksi_wastewater/interlis/interlis_importer_exporter.py:142: InterlisImporterExporterError

```

cymed committed:
add KEK to hierarchy

cymed committed:
add 3d attrs directly

cymed committed:
Merge remote-tracking branch 'upstream/main' into add_3d_export

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
